src/com/facebook/buck/jvm/java/intellij/IjModuleFactory.java

```diff
@@ -359,29 +361,65 @@ public IjModule createModule(
+  private IjModuleJdk getJdk(BuckConfig buckConfig, Iterable<TargetNode<?>> targetNodes, boolean hasAndroid) {
+    ImmutableSortedSet.Builder<String> languageLevelBuilder = new ImmutableSortedSet.Builder<>(Ordering.<String>natural());
+    IntellijConfig intellijConfig = new IntellijConfig(buckConfig);
+    JavaBuckConfig javaBuckConfig = new JavaBuckConfig(buckConfig);
+    JavacOptions defaultJavacOptions = javaBuckConfig.getDefaultJavacOptions();
+    String defaultSourceLevel = defaultJavacOptions.getSourceLevel();
+    String defaultTargetLevel = defaultJavacOptions.getTargetLevel();
+    Optional<String> chosenJdkName = Optional.absent();
+
     for (TargetNode<?> targetNode : targetNodes) {
       BuildRuleType type = targetNode.getType();
-      if (!type.equals(JavaLibraryDescription.TYPE)) {
-        continue;
+      if (hasAndroid && AndroidLibraryDescription.TYPE.equals(type)) {
+        AndroidLibraryDescription.Arg androidArg = (AndroidLibraryDescription.Arg) targetNode.getConstructorArg();
+        if (androidArg.source.isPresent()) {
+          languageLevelBuilder.add(androidArg.source.get());
+        }
+      } else if (!hasAndroid && JavaLibraryDescription.TYPE.equals(type)) {
+        JavaLibraryDescription.Arg javaArg = (JavaLibraryDescription.Arg) targetNode.getConstructorArg();
+        if (javaArg.source.isPresent()) {
+          languageLevelBuilder.add(javaArg.source.get());
+          if (!defaultSourceLevel.equals(javaArg.source.or(defaultSourceLevel)) ||
+              !defaultTargetLevel.equals(javaArg.target.or(defaultTargetLevel))) {
+            chosenJdkName = javaArg.source;
+          }
+        }
       }
+    }
+
+    ImmutableSortedSet<String> languageLevels = languageLevelBuilder.build();
+    String chosenLanguageLevel = !languageLevels.isEmpty() ? languageLevels.first() : defaultSourceLevel;
+    String chosenJdkType;
 
-      JavaBuckConfig javaBuckConfig = new JavaBuckConfig(buckConfig);
-      JavacOptions defaultJavacOptions = javaBuckConfig.getDefaultJavacOptions();
-      String defaultSourceLevel = defaultJavacOptions.getSourceLevel();
-      String defaultTargetLevel = defaultJavacOptions.getTargetLevel();
-      JavaLibraryDescription.Arg arg = (JavaLibraryDescription.Arg) targetNode.getConstructorArg();
-      if (!defaultSourceLevel.equals(arg.source.or(defaultSourceLevel)) ||
-          !defaultTargetLevel.equals(arg.target.or(defaultTargetLevel))) {
-        result = arg.source;
+    if (hasAndroid) {
+      chosenJdkName = intellijConfig.getAndroidSdkName();
+      chosenJdkType = intellijConfig.getAndroidSdkType().or(DEFAULT_ANDROID_SDK_TYPE);
+    } else {
+      if (chosenJdkName.isPresent()) {
+        String jdkName = chosenJdkName.get();
+        if (!jdkName.contains(".")) {
+          chosenJdkName = Optional.of(String.format("1.%s", jdkName));
+        }
       }
+
+      chosenJdkName = chosenJdkName.or(intellijConfig.getJdkName());
```

Comment:
We already have a global language and target level configured in buck: https://buckbuild.com/rule/java_library.html (`source`/`target`) and https://buckbuild.com/concept/buckconfig.html#java.source_level